2024-04-27 13:48:17 - INFO - Username: admin logged in successfully
2024-04-27 13:56:40 - INFO - Username: admin logged in successfully
2024-04-27 13:57:17 - INFO - Username: admin logged in successfully
2024-04-27 13:59:47 - INFO - Username: ellie successfully created an account
2024-04-27 13:59:51 - INFO - Username: admin logged in successfully
2024-04-27 14:00:08 - INFO - Username: admin logged out
2024-04-27 14:00:15 - INFO - Username: ellie logged in successfully
2024-04-27 14:01:17 - INFO - Username: admin logged in successfully
2024-04-27 14:02:14 - INFO - Username: admin logged in successfully
2024-04-27 14:04:01 - INFO - Username: admin logged in successfully
2024-04-28 13:59:34 - INFO - Username: admin logged in successfully
2024-04-28 14:02:20 - INFO - Username: admin logged in successfully
2024-04-28 14:03:26 - INFO - Username: admin logged in successfully
2024-04-28 14:03:36 - INFO - Username: admin logged out
2024-04-28 14:03:43 - INFO - Username: ellie logged in successfully
2024-04-28 14:05:20 - INFO - Username: admin logged in successfully
2024-04-28 14:08:04 - INFO - Username: admin logged in successfully
2024-04-28 14:08:46 - INFO - Username: admin logged in successfully
2024-04-28 14:09:14 - INFO - Username: admin logged in successfully
2024-04-28 14:10:30 - INFO - Username: admin logged in successfully
2024-04-28 14:13:00 - INFO - Username: admin logged in successfully
2024-04-28 14:15:47 - WARNING - Username: admin has a failed login attempt
2024-04-28 14:15:51 - INFO - Username: admin logged in successfully
2024-04-28 14:19:34 - INFO - Username: admin logged in successfully
2024-04-28 14:19:35 - ERROR - Exception on /view_cart [GET]
Traceback (most recent call last):
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 873, in full_dispatch_request
    return self.finalize_request(rv)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 892, in finalize_request
    response = self.make_response(rv)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 1164, in make_response
    raise TypeError(
TypeError: The view function for 'main.view_cart' did not return a valid response. The function either returned None or ended without a return statement.
2024-04-28 14:21:23 - INFO - Username: admin logged in successfully
2024-04-28 14:23:01 - INFO - Username: admin logged in successfully
2024-04-28 14:24:25 - INFO - Username: admin logged in successfully
2024-04-28 14:24:59 - INFO - Username: admin logged in successfully
2024-04-28 14:26:57 - INFO - Username: admin logged in successfully
2024-04-28 14:30:29 - INFO - Username: admin logged in successfully
2024-04-28 14:33:48 - INFO - Username: admin logged in successfully
2024-04-28 14:34:12 - INFO - Username: admin logged in successfully
2024-04-28 14:37:48 - INFO - Username: admin logged in successfully
2024-04-28 14:38:27 - INFO - Username: admin logged in successfully
2024-04-28 14:38:33 - INFO - Username: admin logged out
2024-04-28 14:38:48 - INFO - Username: andrew successfully created an account
2024-04-28 14:39:04 - INFO - Username: ellie logged in successfully
2024-04-28 14:39:21 - INFO - Username: ellie logged out
2024-04-28 14:39:29 - INFO - Username: andrew logged in successfully
2024-04-28 14:39:41 - INFO - Username: andrew logged out
2024-04-28 14:46:58 - INFO - Username: admin logged in successfully
2024-04-28 14:48:20 - INFO - Username: admin logged in successfully
2024-04-28 14:48:58 - INFO - Username: admin logged in successfully
2024-04-28 14:49:26 - INFO - Username: admin logged in successfully
2024-04-28 14:54:21 - INFO - Username: admin logged in successfully
2024-04-28 17:33:09 - INFO - Username: admin logged in successfully
2024-04-28 17:37:50 - INFO - Username: admin logged in successfully
2024-04-28 17:38:34 - INFO - Username: admin logged in successfully
2024-04-28 17:40:17 - INFO - Username: admin logged in successfully
2024-04-28 17:42:03 - INFO - Username: admin logged out
2024-04-28 17:42:09 - INFO - Username: ellie logged in successfully
2024-04-28 17:58:49 - INFO - Username: admin logged in successfully
2024-04-28 18:00:01 - INFO - Username: admin logged out
2024-04-28 18:06:18 - INFO - Username: admin logged in successfully
2024-04-28 18:10:38 - INFO - Username: admin logged in successfully
2024-04-28 18:11:22 - INFO - Username: admin logged in successfully
2024-04-28 18:13:10 - INFO - Username: admin logged in successfully
2024-04-28 18:17:13 - INFO - Username: admin logged in successfully
2024-04-28 18:21:00 - INFO - Username: admin logged in successfully
2024-04-28 18:21:09 - ERROR - Exception on /remove_item/ [POST]
Traceback (most recent call last):
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: remove_item() missing 1 required positional argument: 'asset_id'
2024-04-28 18:21:47 - INFO - Username: admin logged in successfully
2024-04-28 18:23:46 - INFO - Username: admin logged in successfully
2024-04-28 18:24:20 - INFO - Username: admin logged in successfully
2024-04-28 18:26:02 - INFO - Username: admin logged in successfully
2024-04-28 18:26:07 - ERROR - Exception on /remove_item/ [POST]
Traceback (most recent call last):
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: remove_item() missing 1 required positional argument: 'asset_id'
2024-04-28 18:27:00 - INFO - Username: admin logged in successfully
2024-04-28 18:27:14 - ERROR - Exception on /remove_item/ [POST]
Traceback (most recent call last):
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask\app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\ellie\OneDrive\Documents\Programming\Python\Level 6\Major Project\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: remove_item() missing 1 required positional argument: 'asset_id'
2024-04-28 18:28:07 - INFO - Username: admin logged in successfully
